src/client/cl_tempentities.c

```diff
@@ -389,95 +486,78 @@ CL_ParsePlayerBeam(struct model_s *model)
 		MSG_ReadPos(&net_message, offset);
 	}
 
+	if (!model)
```

Comment:
I feel this validation should be above line 474. Just to "keep order", nothing actually terrible.

Reply:
I see what you mean but I put it below the last `MSG_Read` call so that the client message buffer doesn't go out of sync.

Reply:
Hmm, this is actually worth looking at more closely, because if either of the static vars are NULL, it could desync the message buffer. I wonder if there is a way to make this deterministic.

Reply:
I made some interesting discoveries:
* `CL_ParsePlayerBeam` was only used for `TE_HEATBEAM` and `TE_MONSTER_HEATBEAM`, so I renamed it to `CL_ParseHeatBeam`
* The `MSG_ReadByte` in the else clause was never actually reachable because `model` was always either `cl_mod_heatbeam` or `cl_mod_monster_heatbeam`. So I was able to simplify the code

See what you think of the changes in the new commit I just made.

Reply:
The `model = cl_mod_heatbeam;` line to override `cl_mod_monster_heatbeam` looks intentional. Maybe the widow beam model is broken or code for it was never finished. `CL_AddPlayerBeams` only checks for `cl_mod_heatbeam` so even if we restored the widow beam it wouldn't do anything without quite a bit of effort.

Should I just remove the registration for the widow beam entirely?